ui/tests/integration/components/secret-engine/configure-wif-test.js

```diff
@@ -0,0 +1,901 @@
+              `);
+          assert.dom(SES.configureForm).exists(`it lands on the ${type} configuration form`);
+          assert.dom(SES.wif.accessType(type)).isChecked(`defaults to showing ${type} access type checked`);
+          assert.dom(SES.wif.accessType('wif')).isNotChecked('wif access type is not checked');
+          // toggle grouped fields if it exists
+          const toggleGroup = document.querySelector('[data-test-toggle-group]');
+          toggleGroup ? await click(toggleGroup) : null;
+
+          for (const key of expectedConfigKeys(type, true)) {
+            assert
+              .dom(GENERAL.inputByAttr(key))
+              .exists(
+                `${key} shows for ${type} configuration create section when wif is not the access type`
+              );
+          }
+          assert
+            .dom(GENERAL.inputByAttr('issuer'))
+            .doesNotExist(`for ${type}, the issuer does not show when wif is not the access type`);
+        });
+      }
+
+      for (const type of WIF_ENGINES) {
+        test(`${type}: it renders wif fields when user selects wif access type`, async function (assert) {
+          this.id = `${type}-${this.uid}`;
+          this.displayName = allEnginesArray.find((engine) => engine.type === type)?.displayName;
+          this.issuerConfig = createConfig(this.store, this.id, 'issuer');
+          this.config =
+            type === 'aws'
+              ? this.store.createRecord('aws/root-config')
+              : this.store.createRecord(`${type}/config`);
+          this.secondConfig = type === 'aws' ? this.store.createRecord('aws/lease-config') : null;
+          this.config.backend = this.id;
+          this.secondConfig ? (this.secondConfig.backend = this.id) : null;
+          this.type = type;
+
+          await render(hbs`
+                <SecretEngine::ConfigureWif @backendPath={{this.id}} @displayName={{this.displayName}} @type={{this.type}} @model={{this.config}} @issuerConfig={{this.issuerConfig}} @secondModel={{this.secondConfig}}/>
+              `);
+          await click(SES.wif.accessType('wif'));
+          // check for the wif fields only
+          for (const key of expectedConfigKeys(`${type}-wif`, true)) {
+            if (key === 'Identity token TTL') {
+              assert.dom(GENERAL.ttl.toggle(key)).exists(`${key} shows for ${type} wif section.`);
+            } else {
+              assert.dom(GENERAL.inputByAttr(key)).exists(`${key} shows for ${type} wif section.`);
+            }
+          }
+          assert.dom(GENERAL.inputByAttr('issuer')).exists(`issuer shows for ${type} wif section.`);
+        });
+      }
+
+      module('Engine agnostic', function () {
+        /* This module covers code that is the same for all engines. We run them once against one of the engines.*/
```

Comment:
great comment! personally, I'd put it above the module! But that's definitely a nit 😅